cmd/create/machinepool/cmd.go

```diff
@@ -400,6 +419,51 @@ func run(cmd *cobra.Command, _ []string) {
 		}
 	}
 
+	isSpotSet := cmd.Flags().Changed("spot")
+	isSpotMaxPriceSet := cmd.Flags().Changed("spot-max-price")
+
+	spot := args.spotEnabled
+	spotMaxPrice := args.spotMaxPrice
+	if isSpotMaxPriceSet && isSpotSet && !spot {
+		reporter.Errorf("Can't set max price when spot is not enabled")
+		os.Exit(1)
+	}
+
+	if isSpotMaxPriceSet {
+		isSpotSet = true
+		spot = true
+	}
+
+	if !isSpotSet && interactive.Enabled() {
+		spot, err = interactive.GetBool(interactive.Input{
+			Question: "Use spot instances",
+			Help:     cmd.Flags().Lookup("spot").Usage,
+			Default:  spot,
+			Required: false,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid value for spot: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	if spot && !isSpotMaxPriceSet && interactive.Enabled() {
+		spotMaxPrice, err = interactive.GetFloat(interactive.Input{
+			Question: "Spot instance max price",
+			Help:     cmd.Flags().Lookup("spot-max-price").Usage,
+			Required: false,
```

Comment:
You need to set the `Default` here, otherwise you lose whatever value the user already entered

Reply:
We discussed this offline. I don't want to set the default (0) because I don't want to display it to the user. It should be unset if the user doesn't want this.
It's also why we only display it if the user didn't set it earlier

Reply:
No, but I mean specifically if the user sets it. For example:
```
spotMaxPriceInput := interactive.Input{
	Question: "Spot instance max price",
	Help:     cmd.Flags().Lookup("spot-max-price").Usage,
	Required: false,
}
if spotMaxPrice > 0 {
	spotMaxPriceInput.Default = spotMaxPrice
}
```
But you're right that this is unnecessary if we don't display interactive mode once set.